WordPress/Classes/Networking/ReaderPostServiceRemote.m

```diff
@@ -979,6 +983,47 @@ - (NSString *)formatVideoPress:(NSString *)string
     return mstr;
 }
 
+- (NSString *)resizeGalleryImageURLsForContent:(NSString *)content isPrivateSite:(BOOL)isPrivateSite
+{
+    if (!content) {
+        return @"";
+    }
+
+    NSMutableString *mcontent = [content mutableCopy];
+
+    static NSRegularExpression *regexGalleryImages;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        regexGalleryImages = [NSRegularExpression regularExpressionWithPattern:@"<img[^>]*data-orig-file[^>]*\\/>" options:NSRegularExpressionCaseInsensitive error:nil];
+    });
+
+    CGSize imageSize = [UIApplication  sharedApplication].keyWindow.frame.size;
+    CGFloat scale = [[UIScreen mainScreen] scale];
+    CGSize scaledSize = CGSizeApplyAffineTransform(imageSize, CGAffineTransformMakeScale(scale, scale));
+
+    NSArray *matches = [regexGalleryImages matchesInString:mcontent options:NSRegularExpressionCaseInsensitive range:NSMakeRange(0, [mcontent length])];
+    for (NSTextCheckingResult *match in [matches reverseObjectEnumerator]) {
+        NSString *imageElementString = [mcontent substringWithRange:match.range];
+        NSString *srcImageURLString = [self parseValueForAttributeNamed:@"src" inElement:imageElementString];
+        NSString *originalImageURLString = [self parseValueForAttributeNamed:@"data-orig-file" inElement:imageElementString];
+        NSURL *originalURL = [NSURL URLWithString:originalImageURLString];
+        NSURL *modifiedURL;
+
+        if (isPrivateSite) {
+            modifiedURL = [WPImageURLHelper imageURLWithSize:scaledSize forImageURL:originalURL];
+        } else {
+            modifiedURL = [PhotonImageURLHelper photonURLWithSize:imageSize forImageURL:originalURL];
+        }
+
+        [mcontent replaceOccurrencesOfString:srcImageURLString
```

Comment:
This looks risky. I think we could have cases where the same image src appears in multiple places.  This will edit the string in place, perhaps earlier than intended throwing off the matched `NSRange` and impacting the vlaues of `imageElementString` and `srcImageURLString` on the next pass through the loop. 

Maybe just replace the single occurrence rather than all occurrences.